pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1766,18 +1766,32 @@ private void startWorkerService(AuthenticationService authenticationService,
                                     AuthorizationService authorizationService)
             throws Exception {
         if (functionWorkerService.isPresent()) {
-            if (workerConfig.isUseTls() || brokerServiceUrl == null) {
+            if (isNotBlank(brokerServiceUrlTls)) {
                 workerConfig.setPulsarServiceUrl(brokerServiceUrlTls);
             } else {
                 workerConfig.setPulsarServiceUrl(brokerServiceUrl);
             }
-            if (workerConfig.isUseTls() || webServiceAddress == null) {
+            if (isNotBlank(webServiceAddressTls)) {
                 workerConfig.setPulsarWebServiceUrl(webServiceAddressTls);
                 workerConfig.setFunctionWebServiceUrl(webServiceAddressTls);
             } else {
                 workerConfig.setPulsarWebServiceUrl(webServiceAddress);
                 workerConfig.setFunctionWebServiceUrl(webServiceAddress);
             }
+            workerConfig.setTlsEnabled(config.isTlsEnabled());
+            workerConfig.setTlsAllowInsecureConnection(config.isTlsAllowInsecureConnection());
+            workerConfig.setTlsEnableHostnameVerification(config.isTlsHostnameVerificationEnabled());
+            workerConfig.setTlsTrustCertsFilePath(config.getTlsTrustCertsFilePath());
+            workerConfig.setTlsKeyFilePath(config.getTlsKeyFilePath());
+            workerConfig.setTlsCertificateFilePath(config.getTlsCertificateFilePath());
+            workerConfig.setBrokerClientTrustCertsFilePath(config.getBrokerClientTrustCertsFilePath());
+            if (isNotBlank(config.getBrokerClientAuthenticationPlugin())
+                    && isNotBlank(config.getBrokerClientAuthenticationParameters())) {
+                workerConfig.setBrokerClientAuthenticationEnabled(true);
```

Comment:
Why directly enable `BrokerClientAuthentication` when the auth `plugin` and `parameters` are not blank?

It should be left to the user to control, right?

Reply:
yes. these values are configured by user.

Reply:
We should respect `brokerClientAuthenticationEnabled` configuration that is configured by the user, right?